### PR TITLE
Dispatcher Service tests for dispatch workflow module

### LINKED ISSUE #289: Dispatch Service Test - dispatcher_workflow module
Write tests for `refactor/dispatcher/app/core/dispatch_workflow.py`.

### PR DESCRIPTION
**Pull request by FyzHsn**
Fixes #289 

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by FyzHsn** [comment]
@kessler-frost I haven't added the sublattice specific tasks yet. I would like to merge in what I have so far and re-open the issue to add the sublattice tests. The plan is to add these tests once the sublattice integration has been tested out.

**Comment by scottwn** [comment]
looks like an importerror in tests

**Comment by FyzHsn** [comment]
> looks like an importerror in tests

yeah but looking at the list of tests and coverage reports that are getting run, I think there are more problems than that

**Comment by FyzHsn** [comment]
[pre-commit.ci](http://pre-commit.ci/) autofix

**Review by scottwn — APPROVED** [review]

**Review by scottwn — DISMISSED** [review]

File: covalent_dispatcher/_cli/service.py [hunk 1]
```diff
@@ -296,7 +296,11 @@ def _graceful_start(
 
     _rm_pid_file(pidfile)
 
-    pypath = f"PYTHONPATH={UI_SRVDIR}/../tests:$PYTHONPATH" if develop else ""
+    pypath = (
+        f"PYTHONPATH={UI_SRVDIR}/../tests:{UI_SRVDIR}/../covalent/tests:{UI_SRVDIR}/../covalent_dispatcher/tests:{UI_SRVDIR}/../refactor/dispatcher/tests:$PYTHONPATH"
```
Comment (FyzHsn) [inline covalent_dispatcher/_cli/service.py:300]: @wjcunningham7 Does this line look right to you?

**Review by FyzHsn — COMMENTED** [review]
@wjcunningham7 refactor/ui_backend services was giving me errors and so I took it out. I think Alejandro / Val will have to change it.

Some tests in refactor /queuer are also failing - so commented it out.

Two executor and a sync dispatch test also failing.

**Review by kessler-frost — APPROVED** [review]

**Review by scottwn — APPROVED** [review]

**Pull request merged**